zenload/zCMesh.cpp

```diff
@@ -319,7 +321,7 @@ void zCMesh::readObjectData(ZenParser& parser, const std::vector<size_t>& skipPo
 				m_Triangles.reserve(numPolys * 2);
 
 				// Iterate throuh every poly
-				for(int i = 0; i < numPolys; i++)
+				for(auto i = size_t{0}; i < numPolys; i++)
```

Comment:
Interesting initialization. Any resources on why it would be preferred to the "classic" style?

Reply:
I like it because it is consistent with other uses of auto like `auto v = std::vector<int>{1, 2, 3}`. I find it a bit easier read, because on the LHS only the variable name is present, all what is relevant for type/value is on RHS. Having no implicit conversion via the `operator=` is also an advantage.

Here is Herb Sutter's view on it: [GotW #94 Solution: AAA Style (Almost Always Auto)](https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/)

> (b) auto x = type{ init }; when you do want to commit to a specific type by naming a type?
> 
> This is the explicitly typed form, and it still has advantages but they are not as clearly strong as implicitly typed form. The jury is still out on whether to recommend this one wholesale, as we’re still trying it out, but it does offer some advantages and I suggest you try it out for a while and see if it works well for you.
> 
> So here’s the recommendation to consider trying out for yourself: Consider declaring local variables auto x = type{ expr }; when you do want to explicitly commit to a type. (Only when you do want to allow explicit narrowing, use ( ) instead of { }.) The advantages of this typed auto declaration style include:
> 
>      - It guarantees the variable will be initialized; you can’t forget.
>      - It is self-documenting to show that the code is explicitly requesting a conversion.
>      - It won’t allow an accidental implicit narrowing conversion.
>      - It is more symmetric and consistent, both with the basic auto x = init; form and with other parts of C++…
> 
> … which brings us to Question 5.